proxy/src/binary/proxy.rs

```diff
@@ -412,6 +436,23 @@ pub async fn run() -> anyhow::Result<()> {
                     endpoint_rate_limiter.clone(),
                 ));
             }
+
+            // if auth backend is local, we need to load the config file
+            #[cfg(any(test, feature = "testing"))]
+            if let auth::Backend::Local(_) = &auth_backend {
+                // trigger the first config load **after** setting up the signal hook
+                // to avoid the race condition where:
+                // 1. No config file registered when local_proxy starts up
+                // 2. The config file is written but the signal hook is not yet received
+                // 3. local_proxy completes startup but has no config loaded, despite there being a registerd config.
+                let refresh_config_notify = Arc::new(Notify::new());
```

Comment:
this logic doesn't apply here
1. this is before we spawn the signal hook task
2. the refresh_config_notify arc needs to be given to the signal hook task